lib/unified_search_builder.rb

```diff
@@ -22,11 +23,14 @@ def payload
     hash_without_blank_values(
       from: search_params.start,
       size: search_params.count,
-      fields: search_params.return_fields,
+      # `title` and `description` always needed to potentially populate virtual
+      # fields. If not explicitly requested they will not be sent to the user.
+      fields: search_params.return_fields + %w[title description],
       query: query,
       filter: filter,
       sort: sort,
       facets: facets,
+      highlight: highlight,
```

Comment:
highlighting is definitely not a cheap operation.  Could we request it only if the user has asked for the highlighted fields in the response?